superset-core/src/superset_core/api/models.py

```diff
@@ -75,6 +84,83 @@ def backend(self) -> str:
     def data(self) -> dict[str, Any]:
         raise NotImplementedError
 
+    def execute(
+        self,
+        sql: str,
+        options: QueryOptions | None = None,
+    ) -> QueryResult:
+        """
+        Execute SQL synchronously.
+
+        :param sql: SQL query to execute
```

Comment:
One concern I have here: is `sql` a single statement, or can it have multiple statements? I think we need to support both, because some databases don't support sessions, so queries like these must be executed in a single block:

```sql
SET foo=1;  -- needed for the next statement
SELECT * FROM t;
```

But some database drivers don't support multiple statements, so we need to send them one-by-one.